main.js

```diff
@@ -53,64 +55,42 @@ function generateRandomPalette() {
 	}
 }
 
-function updatePalette(array) {
-	for (var i = 0; i < array.length; i++) {
-		allBoxes[i].style.backgroundColor = array[i];
-		allHexes[i].innerText = array[i];
+function updatePalette(colorValues) {
+	for (var i = 0; i < colorValues.length; i++) {
+		allBoxes[i].style.backgroundColor = colorValues[i];
+		allHexes[i].innerText = colorValues[i];
 	}
 }
 
 function savePalette() {
-	var paletteCopy = currentPalette.slice();
-	savedPalettes.push(paletteCopy);
+	var savedPalette = currentPalette.slice();
```

Comment:
Clever, since the original code was using pushing a reference to the current array, not the values. Since currentArray can be changed at any time, when you save a new palette, all of the previous palettes change to the new currentArray, as they are only a reference. In this method, we now push only the values, meaning that they won't change every time currentPalette changes.